src/lib/transition_frontier/transition_frontier.ml

```diff
@@ -1,147 +1,144 @@
+      val of_ledger_builder_diff : Ledger_builder_diff.t -> t
 
-  module Location : Merkle_ledger.Location_intf.S
+      val to_ledger_builder_diff : t -> Ledger_builder_diff.t
+    end
+  end = struct
+    type t = Ledger_builder.Aux.t
 
-  module Ledger_diff : sig
-    type t
-  end
+    let of_ledger_aux = Fn.id
 
-  module Any_base :
-    Merkle_mask.Base_merkle_tree_intf.S
-    with module Addr = Location.Addr
-     and module Location = Location
-     and type account := Account.t
-     and type root_hash := Ledger_hash.t
-     and type hash := Ledger_hash.t
-     and type key := Key.t
-
-  module Ledger_mask : sig
-    include
-      Merkle_mask.Masking_merkle_tree_intf.S
-      with module Addr = Location.Addr
-       and module Location = Location
-       and module Attached.Addr = Location.Addr
-       and type account := Account.t
-       and type location := Location.t
-       and type key := Key.t
-       and type hash := Ledger_hash.t
-       and type parent := Any_base.t
-
-    val merkle_root : t -> Ledger_hash.t
-
-    val apply : t -> Ledger_diff.t -> unit
-
-    val commit : t -> unit
-  end
+    let to_ledger_aux = Fn.id
 
-  module Ledger_database : sig
-    include
-      Merkle_ledger.Database_intf.S
-      with module Location = Location
-       and module Addr = Location.Addr
-       and type account := Account.t
-       and type root_hash := Ledger_hash.t
-       and type hash := Ledger_hash.t
-       and type key := Key.t
-
-    val derive : t -> Ledger_mask.t
-  end
+    (* TODO: Don't hardcode parallelism_log_2 *)
+    let empty = Ledger_builder.Aux.empty ~parallelism_log_2:4
 
-  module Transaction_snark_scan_state : sig
-    type t
+    module Diff = struct
+      type nonrec t = Ledger_builder_diff.t
 
-    module Diff : sig
-      type t
+      let of_ledger_builder_diff = Fn.id
 
-      (* hack until Parallel_scan_state().Diff.t fully diverges from Ledger_builder_diff.t and is included in External_transition *)
-      val of_ledger_builder_diff : Ledger_builder_diff.t -> t
+      let to_ledger_builder_diff = Fn.id
     end
   end
 
+  (* Right now, Staged_ledger is a thin wrapper over Ledger_builder *)
   module Staged_ledger : sig
     type t
 
     val create :
          transaction_snark_scan_state:Transaction_snark_scan_state.t
-      -> ledger_mask:Ledger_mask.t
-      -> t
+      -> ledger:Ledger.Mask.Attached.t
+      -> t Or_error.t
 
     val transaction_snark_scan_state : t -> Transaction_snark_scan_state.t
 
-    val ledger_mask : t -> Ledger_mask.t
+    val ledger : t -> Ledger.Mask.Attached.t
 
-    val apply : t -> Transaction_snark_scan_state.Diff.t -> t Or_error.t
-  end
+    val apply :
+         t
+      -> Transaction_snark_scan_state.Diff.t
+      -> logger:Logger.t
+      -> t Or_error.t
+  end = struct
+    (* TODO *)
+
+    type t = Ledger_builder.t
+
+    let create ~transaction_snark_scan_state ~ledger =
+      Async.Thread_safe.block_on_async_exn (fun () ->
```

Comment:
I don't like this. @bkase explained in person that, pretty soon, @deepthiskumar will have made these things non-async, so this is the temporary glue.